modules/system/classes/MediaLibrary.php

```diff
@@ -834,4 +898,49 @@ protected function generateRandomTmpFolderName($location)
 
         return $tmpPath;
     }
+
+    /**
+     * Generates a incremental file name based
+     * on the existing files in the same folder.
+     *
+     * @param string $path Specifies a file path to check.
+     * @return string The generated file name.
+     */
+    public function generateIncrementedFileName($path): string
+    {
+        $pathInfos = pathinfo($path);
+        $dirName = dirname($this->getMediaPath($path));
+
+        $sameFilesInFolder = Arr::map(
+            $this->getStorageDisk()->files($dirName),
+            function ($value) use ($dirName) {
+                return str_replace($dirName .'/', '', '/'. $value);
```

Comment:
We should probably make use of DIRECTORY_SEPARATOR for these operations. Also, are you able to add unit tests for the new methods?

Reply:
@LukeTowers DIRECTORY_SEPARATOR work nice on Linux but give me a backslash on Windows.
`$this->getStorageDisk()->files($dirName),` give me an array with slashes : 
```php
array:3 [▼
  0 => "media/test-files/file_example_LESS.less"
  1 => "media/test-files/file_example_PDF.pdf"
  2 => "media/test-files/file_example_PDF_1.pdf"
]
```
`$dirName . '/'` = `/media/test-files/` while `$dirName . DIRECTORY_SEPARATOR` = `/media/test-files\`.

That's why I use hardcoded `'/'` because I need to get only files names.

Reply:
@LukeTowers Another solution can be to use `pathinfo` !?

```php
$sameFilesInFolder = Arr::map(
    $this->getStorageDisk()->files($dirName),
    function ($path) {
        $info = pathinfo($path);
        return $info['filename'] .'.'. $info['extension'];
    }
);
```